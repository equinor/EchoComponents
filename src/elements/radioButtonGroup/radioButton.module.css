.header,
.buttonWrapper span {
    font-weight: 500 !important;
}

.buttonWrapper {
    margin: 10px auto 0 auto;
    display: flex;
    max-width: 230px;
    justify-content: space-between;
}

.buttonWrapper label {
    display: flex;
    flex-direction: column;
    font-size: 14px;
    color: black;
}
