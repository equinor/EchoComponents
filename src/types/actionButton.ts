export interface ActionButton {
    title: string;
    onClick: () => void;
    variant: 'contained' | 'outlined' | 'ghost' | 'ghost_icon' | undefined;
}
