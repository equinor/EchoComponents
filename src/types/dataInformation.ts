export interface DataInformation {
    numberOfItems: number;
    label: string;
    ariaLabel?: string;
    onTagInfoClicked: () => void;
    itemType: string;
    itemButtonClassName?: string;
}
