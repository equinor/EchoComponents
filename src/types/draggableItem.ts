export interface DraggableItem {
    id: number;
    element: React.ReactNode;
}
