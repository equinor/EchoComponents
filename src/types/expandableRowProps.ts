import { IconItem } from './iconItem';

export interface ExpandableRowProps {
    iconItems: IconItem[];
    expanded: boolean;
    setExpanded: (status: boolean) => void;
}
