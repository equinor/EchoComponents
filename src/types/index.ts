export * from './actionButton';
export * from './dataInformation';
export * from './draggableItem';
export * from './expandableRowProps';
export * from './iconItem';
export * from './iconListItem';
export * from './linkCardItem';
