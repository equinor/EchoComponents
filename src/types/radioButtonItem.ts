export interface RadioButtonItem {
    title: string;
    isChecked: boolean;
}
