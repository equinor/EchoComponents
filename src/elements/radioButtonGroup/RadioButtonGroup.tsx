import { Radio, Typography } from '@equinor/eds-core-react';
import React, { CSSProperties } from 'react';
import { RadioButtonItem } from '../../types/radioButtonItem';
import styles from './radioButton.module.css';

export interface RadioButtonGroupProps {
    title: string;
    options: RadioButtonItem[];
    style?: CSSProperties;
    onSelected: (index: number) => void;
}

export const RadioButtonGroup: React.FC<RadioButtonGroupProps> = ({
    title,
    options,
    style,
    onSelected
}: RadioButtonGroupProps): JSX.Element => {
    return (
        <div style={style}>
            <Typography variant="body_short" className={styles.header}>
                {title}
            </Typography>
            <div className={styles.buttonWrapper}>
                {options.map((option, index) => {
                    return (
                        <Radio
                            key={index}
                            label={option.title}
                            checked={option.isChecked}
                            onChange={(): void => {
                                onSelected(index);
                            }}
                        />
                    );
                })}
            </div>
        </div>
    );
};
