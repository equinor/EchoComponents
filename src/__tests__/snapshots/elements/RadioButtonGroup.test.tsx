import React from 'react';
import renderer from 'react-test-renderer';
import { RadioButtonGroup } from '../../../elements/radioButtonGroup/RadioButtonGroup';
import { RadioButtonItem } from '../../../types/radioButtonItem';

it('renders RadioButtonGroup correctly ', () => {
    const buttons: RadioButtonItem[] = [
        { title: 'First value', isChecked: false },
        { title: 'Second value', isChecked: true }
    ];
    const tree = renderer
        .create(<RadioButtonGroup title={'My radio button group'} onSelected={jest.fn()} choices={buttons} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
