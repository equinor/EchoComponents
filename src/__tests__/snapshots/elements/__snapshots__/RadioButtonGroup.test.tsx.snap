// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RadioButtonGroup correctly  1`] = `
<div>
  <p
    className="sc-hKgILt dMHJLh header"
  >
    My radio button group
  </p>
  <div
    className="buttonWrapper"
  >
    <label
      className="sc-eltcbb gSlHGN"
      disabled={false}
    >
      <span
        className="sc-gsBrbv cqIwGN"
        disabled={false}
      >
        <input
          checked={false}
          className="sc-jtHMlw foxuLr"
          disabled={false}
          onChange={[Function]}
          type="radio"
        />
        <svg
          aria-hidden={true}
          className="sc-cKZHah"
          fill="var(--eds_interactive_primary__resting, rgba(0, 112, 121, 1))"
          height={24}
          name={null}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="sc-kiYtDG"
            clipRule="evenodd"
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm-5-8a5 5 0 1110 0 5 5 0 01-10 0z"
            fillRule="evenodd"
            name="selected"
          />
          <path
            className="sc-kiYtDG"
            clipRule="evenodd"
            d="M2 12C2 6.48 6.48 2 12 2s10 4.48 10 10-4.48 10-10 10S2 17.52 2 12zm2 0c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8-8 3.58-8 8z"
            fillRule="evenodd"
            name="unselected"
          />
        </svg>
      </span>
      <span
        className="sc-iIEYCM hLQBIp"
      >
        First value
      </span>
    </label>
    <label
      className="sc-eltcbb gSlHGN"
      disabled={false}
    >
      <span
        className="sc-gsBrbv cqIwGN"
        disabled={false}
      >
        <input
          checked={true}
          className="sc-jtHMlw foxuLr"
          disabled={false}
          onChange={[Function]}
          type="radio"
        />
        <svg
          aria-hidden={true}
          className="sc-cKZHah"
          fill="var(--eds_interactive_primary__resting, rgba(0, 112, 121, 1))"
          height={24}
          name={null}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            className="sc-kiYtDG"
            clipRule="evenodd"
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm-5-8a5 5 0 1110 0 5 5 0 01-10 0z"
            fillRule="evenodd"
            name="selected"
          />
          <path
            className="sc-kiYtDG"
            clipRule="evenodd"
            d="M2 12C2 6.48 6.48 2 12 2s10 4.48 10 10-4.48 10-10 10S2 17.52 2 12zm2 0c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8-8 3.58-8 8z"
            fillRule="evenodd"
            name="unselected"
          />
        </svg>
      </span>
      <span
        className="sc-iIEYCM hLQBIp"
      >
        Second value
      </span>
    </label>
  </div>
</div>
`;
